### PR TITLE
CONSOLE-3276: Expose ErrorBoundary capabilities in the SDK

### PR DESCRIPTION
**Pull request by jcaianirh**
Addresses: https://issues.redhat.com/browse/CONSOLE-3276

The console has good error boundary components that are useful for dynamic plugin.
Exposing them will enable the plugins to get the same look and feel of handling react errors as console
The minimum requirement right now is to expose the ErrorBoundaryFallbackPage component from
https://github.com/openshift/console/blob/master/frontend/packages/console-shared/src/components/error/fallbacks/ErrorBoundaryFallbackPage.tsx

Exposed ErrorBoundaryFallbackPage in this pr. We could add additional error boundary components in later pr's if that is something that is desired.

Added a test page in the Demo Dynamic Plugin to test the ErrorBoundaryFallbackPage as well.

Screen Shot 2022-10-14 at 5 22 01 PM

### CODE REVIEW

File: frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts [hunk 1]
```diff
@@ -694,3 +695,29 @@ export const ActionServiceProvider: React.FC<ActionServiceProviderProps> = requi
  */
 export const NamespaceBar: React.FC<NamespaceBarProps> = require('@console/internal/components/namespace-bar')
   .NamespaceBar;
+
+/**
+ * Creates full page ErrorBoundaryFallbackPage component to display the Oh no! Something went wrong.
```
Comment (jhadvig) [inline frontend/packages/console-dynamic-plugin-sdk/src/api/dynamic-core-api.ts:700]: nit
```suggestion
 * Creates full page ErrorBoundaryFallbackPage component to display the "Oh no! Something went wrong."
```

**Review by jhadvig — COMMENTED** [review]
Thanks @jcaianirh ! The changes look good 👍 Adding a few comments.
We should consider to moving this component to the HAC Core SDK eventually.

File: dynamic-demo-plugin/src/components/ErrorBoundary/DemoErrorBoundary.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      error,
+      errorInfo,
+    });
+  }
+
+  render() {
+    const { error, errorInfo } = this.state;
+    const FallbackComponent = this.props.FallbackComponent || DefaultFallback;
+    if (error) {
+      return <FallbackComponent title="Ooops something went wrong in your dynamic plugin" errorMessage={error?.message} componentStack={errorInfo?.componentStack} stack={error?.stack} />;
```
[the remark this answers is not in the document]
Comment (jcaianirh) [reply dynamic-demo-plugin/src/components/ErrorBoundary/DemoErrorBoundary.tsx:32]: Thanks @jhadvig @opayne1

**Comment by jcaianirh** [comment]
Thanks @jhadvig made updates from review if you think its ready please tag.

**Comment by jcaianirh** [comment]
QE Approver:
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @RickJWagner

Console Approver:
/assign @jhadvig

**Review by jhadvig — APPROVED** [review]

**Pull request merged**